pkg/controller/servicebinding/secret.go

```diff
@@ -27,17 +33,90 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// check if payload secret is same as SBR Secret
+func (s *secret) isSame(SBRSecret string) bool {
 	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	// fetch SBR secret
+	resourceClient := s.buildResourceClient()
+	existingSecret, err := resourceClient.Get(
+		SBRSecret,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		logger.Error(err, "Error fetching secret")
+		return false
+	}
+	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+
+	// fetch new payload secret
+	payloadInterim := make(map[string]interface{})
+	for k, v := range s.data {
+		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
+	}
+
+	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
+
+	if comparisonResult.Success {
+		logger.Debug("Secret is same.")
+		return true
+	}
+
+	return false
+}
+
+// secretDataHash creates hash of secret data using sha1
+func secretDataHash(data map[string][]byte) (string, error) {
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	hash := sha1.New()
+	for _, k := range keys {
+		_, err := hash.Write([]byte(k))
+		if err != nil {
+			return "", err
+		}
+		_, err = hash.Write(data[k])
+		if err != nil {
+			return "", err
+		}
+	}
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+func (s *secret) Name() (string, error) {
+	hash, err := secretDataHash(s.data)
+	if err != nil {
+		s.logger.Error(err, "Error creating unique secret name hash")
+		return "", err
+	}
+	if strings.Contains(s.name, hash) {
```

Comment:
Why is this comparison being performed? 

From my understanding, there's no need to do this comparison since the hash has already been computed.

Reply:
This is done because I observed a scenario in which the hash was again appended to the secret name like this:
`sb1-72ddc0c540ab3a290e138726940591debf14c581-72ddc0c540ab3a290e138726940591debf14c581`